xray_vision/mpl_plotting/roi.py

```diff
@@ -42,3 +43,48 @@ def show_label_array(ax, label_array, cmap=None, **kwargs):
                    **kwargs)
 
     return im
+
+
+def show_label_array_on_image(ax, image, label_array, cmap=None, **kwargs):
+    """""
+    This will plot the required ROI's(labeled array on the image
+
+    Additional kwargs are passed through to `ax.imshow`.
+    If `vmin` is in kwargs, it is clipped to minimum of 0.5.
+
+    Parameters
+    ----------
+    ax : Axes
+        The `Axes` object to add the artist too
+
+    image : array
+        The image array
+
+    label_array : array
+        Expected to be an unsigned integer array.  0 is background,
+        positive integers label region of interest
+
+    cmap : str or colormap, optional
+        Color map to use, defaults to 'Paired'
+
+    Returns
+    -------
+    im : AxesImage
+        The artist added to the axes
+
+    im_label : AxesImage
+        The artist added to the axes
+    """
+    if cmap is None:
+        cmap = 'Paired'
+
+    _cmap = copy.copy((mcm.get_cmap(cmap)))
+    _cmap.set_under('w', 0)
+    vmin = max(.5, kwargs.pop('vmin', .5))
+
+    ax.set_aspect('equal')
+    im = ax.imshow(image, cmap='gray', interpolation='none', norm=LogNorm(), **kwargs)
+    im_label = ax.imshow(image*label_array, cmap=cmap,  interpolation='none',
```

Comment:
instead of duplicating all of the label plotting code, just make a call to that function